services/web/server/src/simcore_service_webserver/products/_model.py

```diff
@@ -111,30 +111,40 @@ class Product(BaseModel):
 
     @validator("*", pre=True)
     @classmethod
-    def parse_empty_string_as_null(cls, v):
+    def _parse_empty_string_as_null(cls, v):
         """Safe measure: database entries are sometimes left blank instead of null"""
         if isinstance(v, str) and len(v.strip()) == 0:
             return None
         return v
 
     @validator("name", pre=True, always=True)
     @classmethod
-    def validate_name(cls, v):
+    def _validate_name(cls, v):
         if v not in FRONTEND_APPS_AVAILABLE:
             msg = f"{v} is not in available front-end apps {FRONTEND_APPS_AVAILABLE}"
             raise ValueError(msg)
         return v
 
+    @validator("host_regex", pre=True)
+    @classmethod
+    def _strip_whitespaces(cls, v):
+        if v and isinstance(v, str):
+            # Prevents unintended leading & trailing spaces when added
+            # manually in the database
+            return v.strip()
+        return v
+
     @property
     def twilio_alpha_numeric_sender_id(self) -> str:
         return self.short_name or self.display_name.replace(string.punctuation, "")[:11]
 
     class Config:
         alias_generator = snake_to_camel  # to export
         allow_population_by_field_name = True
+        anystr_strip_whitespace = True
```

Comment:
Q: why you have a validator stripping and this one as well? is this not a doppelganger?

Reply:
Note really. Note that `host_regex` is annotated with `typing.Pattern`  which is not a `str` therefore `anystr_strip_whitespace` does not act onto it. 

To prove that, I have removed the ` @validator("host_regex", pre=True)` and the test i have fails because the string is not parsed away.

I add that to prevent other strings to add extra unecessary whitespaces.